modules/shortcodes/gist.php

```diff
@@ -45,16 +119,34 @@ function github_gist_shortcode( $atts, $content = '' ) {
 
 	$id = ( ! empty( $content ) ) ? $content : $atts[0];
 
-	// Parse a URL
-	if ( ! is_numeric( $id ) ) {
-		$id = preg_replace( '#https?://gist.github.com/([a-zA-Z0-9]+)#', '$1', $id );
+	// Parse a URL to get an ID we can use.
+	$gist_info = jetpack_gist_get_shortcode_id( $id );
+	if ( empty( $gist_info['id'] ) ) {
+		return '<!-- Invalid Gist ID -->';
+	} else {
+		// Add trailing .json to all unique gist identifiers.
+		$id = $gist_info['id'] . '.json';
 	}
 
-	if ( ! $id ) {
-		return '<!-- Invalid Gist ID -->';
+	// The file name can come from the URL passed, or from a shortcode attribute.
+	if ( ! empty( $gist_info['file'] ) ) {
+		$file = $gist_info['file'];
+	} elseif ( ! empty( $atts['file'] ) ) {
+		$file = $atts['file'];
+	} else {
+		$file = '';
+	}
+
+	// Replace - by . to get a real file name from slug.
+	if ( ! empty( $file ) ) {
+		$file = preg_replace( '/\-(?!.*\-)/', '.', $file );
```

Comment:
In theory, `$file` could be very long (if it comes from `$atts['file']`. `strrpos()`/`substr_replace()` might be more performant on very long strings with lots of `-`s.

Reply:
👍 Done in ff8874147e370f5060d59758cb176c53cf9b2c98